### ISSUE TITLE
Be able to instantiate Stats object from the webpack CLI json

### ISSUE THREAD

**Issue by jkanczler** [webpack/webpack#5426]
**Do you want to request a *feature* or report a *bug*?**
I would like to request a feature.

**What is the current behavior?**
I can only use the Stats object from the node.js API, with the compiler and compilation result.

**If the current behavior is a bug, please provide the steps to reproduce.**
It's not a bug.

**What is the expected behavior?**
I would like to see an option to run the webpack CLI with --json flag, and I would like to parse that JSON into an object, without the compilation or the compiler so I can run toString or something like that on the object.

**If this is a feature request, what is motivation or use case for changing the behavior?**
The motivation is to be able to use webpack CLI even if we would like to achieve something like [webpack-vsts-extension](https://github.com/Dealogic/webpack-vsts-extension). In the webpack-vsts-extension to report errors, warnings and the details of the compilation correctly, we had to use webpack node.js API instead of webpack CLI. But sometimes it's kind of handy to use webpack CLI shortcut arguments instead of using webpack config files. See the [reported issue](https://github.com/Dealogic/webpack-vsts-extension/issues/50) in the webpack-vsts-extension.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

**Comment by sokra** [comment]
There is a static `jsonToString` method on the Stats object.

**Comment by jkanczler** [comment]
It was my bad that I came up with the toString example. Sorry for that. Let me rephrase my wish: it would be nice if the CLI --json output basically is a serialized Stats instance, so I can deserialize it from the console output, so I can run all the function of it.

That means basically I would like to see a Stats.js that is decoupled from the compilation or at least it has an other constructor that can accept a JSON as string. Does it make sense?
